types/has/index.d.ts

```diff
@@ -12,5 +12,5 @@
  *
  * @throws {TypeError} If `target` is nullish.
  */
-declare function hasOwnProperty(target: {}, property: PropertyKey): boolean;
+declare function hasOwnProperty<P extends PropertyKey>(target: {}, property: P): target is { [K in P]: any };
```

Comment:
```suggestion
declare function hasOwnProperty<P extends PropertyKey>(target: {}, property: P): target is { [K in P]: unknown };
```

Reply:
That won’t work in **TypeScript 2.x**.

Reply:
Neither will predicate syntax.

Reply:
Type predicate syntax is supported in **TypeScript 2.8**, otherwise the tests wouldn’t have passed.

Reply:
ah, fair enough. there's a way to specify a base set of types for TS 2, and then a better set of types for TS 3; could we do that? `any` is a plague.

Reply:
DT only supports 3.0 and above as of May, so you can use mapped types and unknown for everything.

Reply:
hooray! looking forward to the PR that changes every `any` in the entire repo to `unknown`